// module/websocketserver.js
/**
 * Web Socket io server
 */
var sio = require('socket.io');
var MongoClient = require('mongodb').MongoClient;
var User = function(client){
	this.name = null ;
	this.client = client ;
	this.session = client.sessionId;
	this.activity = new Date().getTime();
	this.status = "active";
	this.room = null ;
	this.send = function(data){
		this.client.send(data);
	};
};
var DB = {
	init:function(dbUrl){
		MongoClient.connect("mongodb://localhost:27017/"+dbUrl, function(err, db) {
			  if(err) { 
				  return console.dir(err); 
			   }
			});
	},
	query:function(db,data){
		
	},
	insert:function(db, data){
		
	},
	update:function(db,data){
		
	},
};
var Commands = {
	close: function(room,data){
		data = {command:"close",
		        message: data.message};
		console.log(data.message);
	},
	
	login: function(data){
		
	},
	parse: function(user,data){
		if(data.hasOwnProperty("command")){
			console.log("has command");
		}
		
		if(Commands.commands.hasOwnProperty(data.command)){
			console.log("command:"+data.command);
			Commands.commands[data.command](user,data);
		}else{
			console.log("command:Unknow command");
		}
		
		
	},
	commands: {
		login:function(user,data){
			var username = data.username;
			var password = data.password;
			console.log("username:"+username+";password:"+password);
		},
		chat:function(user,data){
		},
		leave:function(user,data){
		},
		register:function(user,data){
			var username = data.username ;
			var password = data.password ;
			
		}
	}
	
};

module.exports = function(app){
	DB.init("demo1");
	this.io = sio.listen(app);
	console.log("socket io start listen...");
	this.io.sockets.on('connection',function(client){
			client.user = new User(client);
			client.on("message",function(data){
				Commands.parse(client.user,data);
			});
			client.on("disconnect",function(){
				console.log("clinet disconnected...");
			});
		}
	);
	
}
